src/cc_toolchain/BUILD

```diff
@@ -0,0 +1,257 @@
+# -*- python -*-
+
+# Copyright 2018-2019 Josh Pieper, jjp@pobox.com.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package(default_visibility = ["//visibility:public"])
+
+load(":cc_toolchain_config.bzl", "cc_toolchain_config", "cc_toolchain_config_stm32")
+
+# Create environments for each CPU this toolchain supports
+environment(name = "k8")
+
+environment(name = "stm32f0")
+
+environment(name = "stm32f4")
+
+environment(name = "stm32g4")
+
+environment(name = "stm32h7")
+
+environment_group(
+    name = "cpus",
+    defaults = [
+        ":k8",
+        ":stm32h7",
+    ],
+    environments = [
+        ":k8",
+        ":stm32h7",
+    ],
+)
+
+# This represents a mapping of CPU -> Compiler To Use
+# The CPU can be chosen by using `--cpu` as a bazel flag (ex. `--cpu=stm32h7`)
+# otherwise it will default to using whatever CPU is in the system doing the compiling
+cc_toolchain_suite(
+    name = "toolchain",
+    toolchains = {
+        # k8 is any x86 system
+        "k8": "cc_toolchain_linux",
+        "k8|compiler": "cc_toolchain_linux",
+        # stm32f0 MCU
+        "stm32f0": "cc_toolchain_stm32f0",
+        "stm32f0|compiler": "cc_toolchain_stm32f0",
+        # stm32f4 MCU
+        "stm32f4": "cc_toolchain_stm32f4",
+        "stm32f4|compiler": "cc_toolchain_stm32f4",
+        # stm32g4 MCU
+        "stm32g4": "cc_toolchain_stm32g4",
+        "stm32g4|compiler": "cc_toolchain_stm32g4",
+        # stm32h7 MCU
+        "stm32h7": "cc_toolchain_stm32h7",
+        "stm32h7|compiler": "cc_toolchain_stm32h7",
+    },
+)
+
+filegroup(
+    name = "empty",
+    srcs = [],
+)
+
+filegroup(
+    name = "arm_everything",
+    srcs = [
+        "@arm_developer_gcc//:everything",
+    ] + glob(["wrapper/arm-none-eabi-*"]),
+)
+
+filegroup(
+    name = "clang_wrappers",
+    srcs = glob(["wrapper/clang-*"]) + [
+        "@llvm_clang//:binaries",
```

Comment:
Are binaries part of the wrappers?

Reply:
How this is organized actually doesn't make a ton of sense. 

- moved `@llvm_clang//:binaries` to `clang_all`
- moved `clang_wrappers` declaration into it's own build file in the `wrapper` folder